lightning/backend/sql2kv.go

```diff
@@ -179,6 +180,9 @@ func (kvcodec *tableKVEncoder) Encode(
 			return nil, logKVConvertFailed(logger, row, j, col.ToInfo(), err)
 		}
 		record = append(record, value)
+		if isAutoIncCol {
+			kvcodec.tbl.RebaseAutoID(kvcodec.se, value.GetInt64(), false)
```

Comment:
This `RebaseAutoID` is called at every row in my understanding, why could this 
rebase-per-row gives tighter upper bound than rebase-per-chunk, I think they will eventually reach almost same auto-id and now it's more expensive.

Reply:
The change allowing a tighter bound is deleting

https://github.com/pingcap/tidb-lightning/blob/4eb74ec2a454e10c09113d7da89394cc22c417eb/lightning/restore/restore.go#L649-L653

since `RowIDMax` is normally much larger than the actual row number. 

Yes this could become more expensive, but we should perform a measurement to ensure. I bet the CAS loop is still much cheaper than the actual KV encoding.

Reply:
Now i see rebased value is cast from `rowId`, which is smaller or equal than row's content. And I agree CAS is lightweight. I don't know if we could move this rebase-per-row outside, such as on [chunk's EOF](https://github.com/pingcap/tidb-lightning/blob/master/lightning/restore/restore.go#L1660), so there's fewer repeat.

Reply:
CAS is lightweight so maybe we shouldn't bother move it outside ⬆️